mmedit/structures/edit_data_sample.py

```diff
@@ -238,3 +220,104 @@ def gt_label(self, value: LabelData):
     @gt_label.deleter
     def gt_label(self):
         del self._gt_label
+
+    @classmethod
+    def stack(cls,
+              data_samples: Sequence['EditDataSample']) -> 'EditDataSample':
+        """Stack a list of data samples to one. All tensor fields will be
+        stacked at first dimension. Otherwise the values will be saved in a
+        list.
+
+        Args:
+            data_samples (Sequence['EditDataSample']): A sequence of
+                `EditDataSample` to stack.
+
+        Returns:
+            EditDataSample: The stacked data sample.
+        """
+        # 1. check key consistency
+        keys = data_samples[0].keys()
+        assert all([data.keys() == keys for data in data_samples])
+
+        meta_keys = data_samples[0].metainfo_keys()
+        assert all(
+            [data.metainfo_keys() == meta_keys for data in data_samples])
+
+        # 2. stack data
+        stacked_data_sample = EditDataSample()
+        for k in keys:
+            values = [getattr(data, k) for data in data_samples]
+            # 3. check type consistent
+            value_type = type(values[0])
+            assert all([type(val) == value_type for val in values])
+
+            # 4. stack
+            if isinstance(values[0], torch.Tensor):
+                stacked_value = torch.stack(values)
+            elif isinstance(values[0], LabelData):
+                labels = [data.label for data in values]
+                values = torch.stack(labels)
+                stacked_value = LabelData(label=values)
+            else:
+                stacked_value = values
+            stacked_data_sample.set_field(stacked_value, k)
+
+        # 5. stack metainfo
+        for k in meta_keys:
+            values = [data.metainfo[k] for data in data_samples]
+            stacked_data_sample.set_metainfo({k: values})
+
+        stacked_data_sample._is_stacked = True
+        return stacked_data_sample
+
+    def split(self) -> Sequence['EditDataSample']:
```

Comment:
maybe we can assert the value of `_is_stacked` before splitting. If `_is_stacked=False`, we may raise a warning, like 'You are splitting an EditDataSample that has not been stacked before'.